src/Arcus.WebApi.Security.Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -14,18 +14,50 @@ public class JwtTokenReader : IJwtTokenReader
             "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration";
 
         private readonly string _applicationId;
+        private readonly TokenValidationParameters _tokenValidationParameters;
         private readonly JwtSecurityTokenHandler _handler;
         private readonly ConfigurationManager<OpenIdConnectConfiguration> _configManager;
 
-        public JwtTokenReader(string applicationId)
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
+        /// <param name="applicationId">Azure AD Application used as audience to validate against</param>
+        public JwtTokenReader(string applicationId) : this()
         {
             Guard.NotNullOrWhitespace(applicationId, nameof(applicationId));
 
             _applicationId = applicationId;
+            _configManager = new ConfigurationManager<OpenIdConnectConfiguration>(MicrosoftDiscoveryEndpoint, new OpenIdConnectConfigurationRetriever());
+        }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
+        /// <param name="tokenValidationParameters">Collection of parameters to influence how the token validation is done</param>
+        public JwtTokenReader(TokenValidationParameters tokenValidationParameters) : this(MicrosoftDiscoveryEndpoint, tokenValidationParameters)
+        {
+        }
 
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="openIdConnectDiscoveryUri">Uri of an OpenId connect endpoint for discovering the configuration</param>
+        /// <param name="tokenValidationParameters">Collection of parameters to influence how the token validation is done</param>
+        public JwtTokenReader(string openIdConnectDiscoveryUri, TokenValidationParameters tokenValidationParameters) :
```

Comment:
Don't know if it makes sense to swap the parameters on this ctor ?  What's your idea on this @stijnmoreels ?

Reply:
Good idea

Reply:
Yes, I agree with @fgheysels . It's better to gradually show overloads with the same parameters in the beginning. Better discovery.
But yeah, already handled and fixed, so what am I doing here 🤡 .

Reply:
Already done ;)